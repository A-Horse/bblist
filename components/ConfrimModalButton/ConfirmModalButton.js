import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'components/widget/Modal/Modal';
import { Button } from 'components/widget/Button/Button';
import { isEnterKey } from 'utils/keyboard';

import './ConfirmModalButton.scss';

export class ConfrimModalButton extends Component {
  static propTypes = {
    onConfirm: PropTypes.func.isRequired
  };

  state = {
    toggle: false
  };

  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
    this.onModalClose = this.onModalClose.bind(this);
    this.onConfirm = this.onConfirm.bind(this);
    this.onKeyDown = this.onKeyDown.bind(this);
  }

  buildClassName() {
    return `confrim-modal-button ${this.props.className ? ' ' + this.props.className : ''}`;
  }

  onClick() {
    this.setState({ toggle: true });
  }

  onModalClose() {
    this.setState({ toggle: false });
  }

  onConfirm() {
    this.props.onConfirm();
    this.setState({ toggle: false });
  }

  onKeyDown(event) {
    if (isEnterKey(event)) {
      this.onConfirm();
    }
  }

  render() {
    return (
      <div className={this.buildClassName()} onClick={this.onClick}>
        {this.props.children}
        <Modal
          toggle={this.state.toggle}
          close={this.onModalClose}
          className="confirm-button-modal"
          onKeyDown={this.onKeyDown}
        >
          <div>
            <div>Are you sure you want to delete it?</div>
            <div className="confrim-modal-button--operation">
              <Button
                className="confirm-button"
                type="submit"
                styleType="dangerous"
                onClick={this.onConfirm}
              >
                Confrim
              </Button>
              <Button className="cancel-button" onClick={this.onModalClose}>
                Cancel
              </Button>
            </div>
          </div>
        </Modal>
      </div>
    );
  }
}

export default ConfrimModalButton;
