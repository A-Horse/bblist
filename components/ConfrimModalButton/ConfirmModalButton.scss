@import 'style/variable';

.confirm-button-modal {
  border-radius: 5px;
  top: 40%;
  left: 50%;
  transform: translate(-50%, -50%);

  .confrim-modal-button--operation {
    margin: auto;
    margin-top: 20px;
    width: 80%;
    display: flex;
    justify-content: space-between;
  }
}
