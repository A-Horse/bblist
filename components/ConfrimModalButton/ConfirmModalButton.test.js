import React from 'react';
import ConfirmModalButton from './ConfirmModalButton';
import renderer from 'react-test-renderer';

test('ConfirmModalButton', () => {
  const onConfirm = jest.fn();
  const tree = renderer.create(
    <ConfirmModalButton className="confirm-modal-button" onConfirm={onConfirm}>
      ConfirmModalButton
    </ConfirmModalButton>
  );
  expect(tree).toMatchSnapshot();
});
