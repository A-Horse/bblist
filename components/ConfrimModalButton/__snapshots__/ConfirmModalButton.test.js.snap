// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModalButton 1`] = `
<div
  className="confrim-modal-button  confirm-modal-button"
  onClick={[Function]}
>
  ConfirmModalButton
  <noscript />
</div>
`;
